WhisperAPI/WhisperAPI/Services/InMemoryContexts.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using WhisperAPI.Models;
+
+namespace WhisperAPI.Services
+{
+    public class InMemoryContexts : IContexts
+    {
+        private object _removeOldLock = new object();
+
+        public InMemoryContexts(TimeSpan contextLifeSpan)
+        {
+            this.ContextLifeSpan = contextLifeSpan;
+            this.ConversationContexts = new Dictionary<Guid, ConversationContext>();
```

Comment:
Use a ConcurrentDictionary instead which is thread-safe. You'll see the methods are a bit different.